pkg/controller/workspace/provision/pvc.go

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package provision
+
+import (
+	"strings"
+
+	"github.com/che-incubator/che-workspace-operator/pkg/controller/workspace/prerequisites"
+
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IsPVCRequired checks to see if we need a PVC for the given devfile. If cloud-shell is present in the devfile,
+// we do not need PVC else we do
+func IsPVCRequired(components []v1alpha1.ComponentSpec) bool {
+	for _, comp := range components {
+		if strings.Contains(comp.Id, config.CloudShellID) && comp.Type == v1alpha1.CheEditor {
```

Comment:
I think it would be a bit smarter if we check volumes in podsAdditional instead of sticking to component id...

Reply:
Here is my draft proposal https://github.com/sleshchenko/che-workspace-operator/commit/91206b47e32c761ce1bfa79edc0d5ad1abf27384
Maybe we even should get rid of object.go#SyncObject because each object type should be sync in different way: like PVC is immutable, like you can't assign new size.
While probably we should update Roles and RolesBindings...

Reply:
I've just been debugging this for a bit and it looks like podAdditions.Volumes always has length 0 for some reason meaning the PVC will never be created.

If we compare the volumeMounts to the PVCVolumeName it works though:
```go
func IsPVCRequired(components []v1alpha1.ComponentDescription) bool {
	volumeName := config.ControllerCfg.GetWorkspacePVCName()
	for _, comp := range components {
		for _, cont := range comp.PodAdditions.Containers {
			for _, vm := range cont.VolumeMounts {
				if vm.Name == volumeName {
					return true
				}
			}
		}
	}
	return false
}
```

Reply:
:+1: on checking for the pvc volume mount.

The reason you're not finding it in `podAdditions.Volumes` is because we currently handle it separately -- otherwise we'd have to deduplicate the list of `podAdditions.Volumes`, which would prevent us catching issues like two components trying to add the same volume.